sous_chef/tests/unit_tests/menu/test_create_menu.py

```diff
@@ -1,66 +1,83 @@
 import pandas as pd
 import pytest
 from freezegun import freeze_time
 from hydra import compose, initialize
+from pandera.typing import DataFrame
 from sous_chef.formatter.ingredient.format_ingredient import (
     Ingredient,
     IngredientFormatter,
 )
-from sous_chef.menu.create_menu import Menu, MenuIngredient, MenuRecipe
+from sous_chef.menu.create_menu import (
+    Menu,
+    MenuIngredient,
+    MenuRecipe,
+    MenuSchema,
+)
 from sous_chef.messaging.gsheets_api import GsheetsHelper
-from tests.unit_tests.util import assert_equal_dataframe, create_recipe
+from tests.unit_tests.util import create_recipe
+from tests.util import assert_equal_dataframe, assert_equal_series
 
 FROZEN_DATE = "2022-01-14"
 
 
-def create_menu_row(
-    weekday: str = "Friday",
-    meal_time: str = "dinner",
-    item_type: str = "recipe",
-    eat_factor: float = 1.0,
-    eat_unit: str = "",
-    # gsheets has "", whereas read_csv defaults to np.nans
-    freeze_factor: Union[float, str] = "",
-    item: str = "dummy",
-    grocery_list: str = "Y",
-    menu_list: str = "Y",
-):
-    return pd.Series(
-        {
-            "weekday": weekday,
-            "meal_time": meal_time,
-            "eat_factor": eat_factor,
-            "eat_unit": eat_unit,
-            "freeze_factor": freeze_factor,
-            "item": item,
-            "type": item_type,
-            "grocery_list": grocery_list,
-            "menu_list": menu_list,
-        }
-    )
+@dataclass
+class MenuBuilder:
+    menu: pd.DataFrame = None
+
+    def add_row(
+        self,
+        weekday: str = "Friday",
+        meal_time: str = "dinner",
+        item_type: str = "recipe",
+        eat_factor: float = 1.0,
+        # gsheets has "", whereas read_csv defaults to np.nans
+        eat_unit: str = "",
+        freeze_factor: float = 0.0,
+        item: str = "dummy",
+    ) -> pd.Series:
+        row = DataFrame[MenuSchema](
+            {
+                "weekday": weekday,
+                "meal_time": meal_time,
+                "eat_factor": eat_factor,
+                "eat_unit": eat_unit,
+                "freeze_factor": freeze_factor,
+                "item": item,
+                "type": item_type,
+            },
+            index=[0],
+        )
+
+        if self.menu is None:
+            self.menu = row
+        else:
+            self.menu = pd.concat([self.menu, row], ignore_index=True)
+        return row.squeeze().copy(deep=True)
+
+    def get_menu(self) -> pd.DataFrame:
+        return self.menu
 
 
 @pytest.fixture
-def menu_default():
-    return pd.concat(
-        [
-            create_menu_row(item="recipe_no_freezing"),
-            create_menu_row(item="recipe_with_freezing", freeze_factor=0.5),
-            create_menu_row(
-                item="manual ingredient",
-                item_type="ingredient",
-                eat_factor=1,
-                eat_unit="pkg",
-            ),
-        ],
-        axis=1,
-    ).T
+def menu_default(menu):
```

Comment:
In addition, could have another fixture here similar to as done with RecipeBookBuilder, which just initializes a default MenuBuilder. Something like:
```
@pytest.fixture
def menu_builder():
    return MenuBuilder()
```
That fixture can then be passed in the various tests and `menu_builder = MenuBuilder()` can be omitted. What do you think?